src/main/java/com/aws/greengrass/lifecyclemanager/KernelLifecycle.java

```diff
@@ -265,94 +266,55 @@ void initConfigAndTlog(String configFilePath) {
-                    // Attempt to load from backup tlog file
-                    Path backupTlogPath = CommitableFile.getBackupFile(transactionLogPath);
-                    boolean backupValid = false;
-                    if (Files.exists(backupTlogPath)) {
-                        try {
-                            ConfigurationReader.validateTlog(backupTlogPath);
-                            backupValid = true;
-                        } catch (IOException e) {
-                            logger.atError().log("Backup transaction log at {} is invalid", backupTlogPath, e);
-                        }
-                    }
-
-                    if (backupValid) {
-                        logger.atError()
-                                .log("Transaction log {} is invalid and so is the backup at {}, will attempt to "
-                                                + "load configuration from {}", transactionLogPath, backupTlogPath,
-                                        bootstrapTlogPath, tlogValidationError);
-                        kernel.getConfig().read(backupTlogPath);
-                        readFromNonTlog = true;
-                    } else if (bootstrapTlogExists) {
-                        // If no backup or if the backup was invalid, then try loading from bootstrap
-                        logger.atError()
-                                .log("Transaction log {} is invalid and no usable backup exists, will attempt to load "
-                                                + "configuration from {}", transactionLogPath, bootstrapTlogPath,
-                                        tlogValidationError);
-                        kernel.getConfig().read(bootstrapTlogPath);
-                        readFromNonTlog = true;
-                    } else {
-                        // There are no files to load from
-                        logger.atError()
-                                .log("Transaction log {} is invalid and no usable backup exists", transactionLogPath,
-                                        tlogValidationError);
-                    }
+                // read from external configs
+                Path externalConfig = nucleusPaths.configPath().resolve(Kernel.DEFAULT_CONFIG_YAML_FILE_READ);
+                boolean externalConfigFromCmd = Utils.isNotEmpty(kernelCommandLine.getProvidedInitialConfigPath());
+                if (externalConfigFromCmd) {
+                    externalConfig = Paths.get(kernelCommandLine.getProvidedInitialConfigPath());
                 }
-
+                // not validating its content since the file could be in non-tlog format
                 boolean externalConfigExists = Files.exists(externalConfig);
                 // If there is no tlog, or the path was provided via commandline, read in that file
-                if ((externalConfigFromCmd || !tlogExists) && externalConfigExists) {
+                if ((externalConfigFromCmd || !transactionTlogValid) && externalConfigExists) {
```

Comment:
Do we really need `externalConfigFromCmd` (as long as `externalConfigExists` is true, we should read from `externalConfig`) ?

Reply:
It's not as long as `externalConfigExists` is true

Only when 1) there is no valid `conifg.tlog` or 2) the commandline provides a path using the `--init-config` option

Reply:
Talked offline. Seems like we read both from `config.tlog` and `externalConfig` when `transactionTlogValid` is true. If that is the intended behavior then this check makes sense to me.